### PR TITLE
Subscription feature

### PR DESCRIPTION
**Pull request by armstrtw**
add subscribe function.

passes R CMD check.

### CODE REVIEW

File: man/subscribe.Rd [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+\value{
+This function always returns NULL.
+}
+\description{
+This function uses the Bloomberg API to stream live market data
+}
+\examples{
+\dontrun{
+  subscribe(securities=c("/ticker/TYZ5 Comdty","/cusip/912810RE0@BGN"),
```
Comment (eddelbuettel) [inline man/subscribe.Rd]: Never seen this notation before. Is that described somewhere in the copious Bloomberg docs?

File: src/subscribe.cpp [hunk 2]
```diff
@@ -0,0 +1,230 @@
+// -*- mode: C++; c-indent-level: 4; c-basic-offset: 4; indent-tabs-mode: nil; -*-
+//
+//  subscribe.cpp -- "Bloomberg Data Point" query function for the BLP API
```
Comment (eddelbuettel) [inline src/subscribe.cpp]: Looks like the RHS needs an update :)

Comment (armstrtw) [inline src/subscribe.cpp]: RHS?

I'll fix the file. it's an artifact of copying from bdp.cpp.

File: man/subscribe.Rd (hunk 1 above)
Comment (armstrtw) [inline man/subscribe.Rd]: from blpapi_subscriptionlist.h

you think I should add some of this documentation to the man page?

// STRUCTURE OF SUBSCRIPTION STRING
// ---------------------------------
// The simplest form of a subscription string is a *fully* *qualified*
// subscription string, which has the following structure:
//
// "//blp/mktdata/ticker/IBM US Equity?fields=BID,ASK&interval=2"
//  \-----------/\------/\-----------/\------------------------/
//        |          |         |                  |
//     Service    Prefix   Instrument           Suffix

File: src/subscribe.cpp (hunk 2 above)
Comment (eddelbuettel) [inline src/subscribe.cpp]: (RHS == right-hand side)
And thanks.

Comment (armstrtw) [inline src/subscribe.cpp]: so, how do I refresh the PR?  cancel it and submit a new one?

File: man/subscribe.Rd (hunk 1 above)
Comment (eddelbuettel) [inline man/subscribe.Rd]: Yes -- I think that would not hurt.

Comment (armstrtw) [inline man/subscribe.Rd]: github knows your intentions before you do... spooky.

Comment (armstrtw) [inline man/subscribe.Rd]: you think a straightup copy/paste/reformat from the c++ header would be ok?

Comment (eddelbuettel) [inline man/subscribe.Rd]: Could Bloomberg be offended if we copied?  If there is something online, a URL may do.  Else a reference to the header?

Comment (armstrtw) [inline man/subscribe.Rd]: that's a good point. I think I can produce a link.

**Comment by johnlaing** [comment]
I don't know that we need to tie an exposition of this notation to this particular feature. AFAIK, this structure is common across the entire API. E.g.:
```r
jlaing@xenon:~$ R
> require(Rblpapi)
Loading required package: Rblpapi
> blpConnect(host="app", port=8194)
> bdp("/ticker/TYZ5 Comdty", "PX_LAST")
                    PX_LAST
/ticker/TYZ5 Comdty     126
```

**Comment by armstrtw** [comment]
how does this sit with everyone:
##' @details
##' The subscribe function allows one to subscribe to streaming market
##' quotes.
##'
##' Full detials of the subscription string can be found in the
##' header file blpapi_subscriptionlist.h at the following link:
##' http://static.bloomberglabs.com/api/cpp/doc/3.8/blpapi__subscriptionlist_8h_source.html

**Comment by eddelbuettel** [comment]
@johnlaing Yes, thanks, once Whit showed the reference to the header I suspected something like that.

@armstrtw Is it also possible to drop the leading `/ticker`, ie could one subsribe to `IBM US Equity` or any odd future?

The link looks fine.  That is a nice balance between a brief synopsis and further reference.

**Comment by armstrtw** [comment]
yes, I think the ticker can be dropped from the TY example.  let me check.

**Comment by armstrtw** [comment]
yes, it works without the '/ticker' prefix.  I updated the futures example.  I left the cusip example in there for variety.

**Comment by armstrtw** [comment]
Thanks.  do you normally version bump for a new feature? or just leave it as a separate commit?

**Comment by eddelbuettel** [comment]
Yes, a fourth digit +1 may be on order as are ChangeLog and NEWS entries.

**Pull request merged**